BRCA_Deep_Dive.Rmd

```diff
@@ -0,0 +1,1511 @@
+    start = "1979-01-01",
+    end = "2023-12-31"
+    ) %>%
+  # convert daily values for each grid cell to daily mean for all cells
+  dplyr::mutate(CF = "Historical")
+                                        
+
+# Get future climate data for water source area
+clim_source_fut <- 
+  get_climate_future(
+    sf = watersupply_watershed,
+    col_name = "name",
+    start = "2006-01-01",
+    end = "2070-01-01",
+    GCM = select_cfs$GCM
+    ) %>%
+  dplyr::mutate(GCM = paste0(GCM,'.',RCP)) %>%
+  left_join(select_cfs %>% dplyr::select(c("GCM","CF")), by = c("GCM")) %>%
+  dplyr::select(-c("GCM","Ensemble","RCP")) 
+
+# Bias correction
+ggplot() +
+  geom_line(data = clim_source_fut %>% dplyr::filter(year(date) < 2023), 
+            aes(x = date, y = tmean_C, color = "MACA")) +
+  geom_line(data = clim_source_hist %>% dplyr::filter(year(date) > 2005),
+            aes(x = date, y = tmean_C, color = "GridMET"))
+
+# Get historical climate data for nearest NWIS stream gage
+clim_nwis <- 
+  get_climate_historic(
+    sf = nwis_nearest_watershed,
+    col_name = "site_no",
+    start = nwis_nearest_discharge$date %>% min(),
+    end = "2023-01-01") %>%
+  dplyr::mutate(CF = "Historical") %>%
+  # Join wbm results with discharge
+  left_join(., nwis_nearest_discharge , by = c("date", "site_no")) 
+
+
+
+# Get unique points from each climate dataset
+source_fut_pts <- clim_source_fut %>% 
+  ungroup() %>% 
+  distinct(x,y, .keep_all = TRUE) %>% 
+  st_as_sf(., coords = c("x","y"), crs = st_crs(4326), remove = FALSE) %>%
+  dplyr::select(c(date,CF,name,x,y))
+
+source_hist_pts <- clim_source_hist %>% 
+  ungroup() %>% 
+  distinct(x,y, .keep_all = TRUE) %>% 
+  st_as_sf(., coords = c("x","y"), crs = st_crs(4326), remove = FALSE) %>%
+  dplyr::select(c(date,CF,name,x,y))
+
+nwis_pts <- clim_nwis %>% 
+  ungroup() %>% 
+  distinct(x,y, .keep_all = TRUE) %>% 
+  st_as_sf(., coords = c("x","y"), crs = st_crs(4326), remove = FALSE) %>%
+  dplyr::select(c(date,CF,site_no,x,y))
+
+# Plot
+mapview(source_fut_pts, 
+        col.regions = "black",
+        layer.name = "MACA") + 
+  mapview(watersupply_watershed, 
+          col.regions = "dodgerblue", 
+          alpha.regions = 0.2,
+          layer.name = "Source Watershed") + 
+  mapview(source_hist_pts, 
+          col.regions = "red",
+          layer.name = "GridMET") +
+mapview(source_hist_pts %>% 
+          st_transform(., 32612) %>% 
+          st_buffer(, dist = 2000),
+        col.regions = "red",
+        alpha.regions = 0.2,
+        legend = FALSE) +
+  mapview(nwis_pts,
+          col.regions = "red",
+          legend = FALSE) +
+  mapview(nwis_nearest_watershed,
+          col.regions = "seagreen",
+          alpha.regions = 0.3,
+          layer.name = "NWIS Watershed")
+
+```
+
+
+
+### 9. WBM 
+
+Run the NPS water balance model for each gridmet point in the AOI
+
+```{r run_wbm, eval = TRUE, echo = FALSE, message = FALSE, warning = FALSE}
+
+source("src/WB_Functions.R")
```

Comment:
Does this not load with the setup script...?

Reply:
It does now! Removed.